### PR TITLE
nvme: Align with recent libnvme additions and drop the discovery functionality

### PR DESCRIPTION
**Pull request by tbzatek**
Bunch of changes geared towards the 3.0 release. Please see respective commit messages for details.

### CODE REVIEW

**Review by vojtechtrefny — APPROVED** [review]
Looks good to me.

File: configure.ac [hunk 1]
```diff
@@ -238,7 +238,12 @@ AS_IF([test "x$with_nvdimm" != "xno"],
       [])
 
 AS_IF([test "x$with_nvme" != "xno"],
-      [LIBBLOCKDEV_PKG_CHECK_MODULES([NVME], [libnvme >= 1.2])],
+      [LIBBLOCKDEV_PKG_CHECK_MODULES([NVME], [libnvme >= 1.2])
+      AS_IF([$PKG_CONFIG --atleast-version=1.3 libnvme],
+            [AC_DEFINE([HAVE_LIBNVME_1_3])], [])
```
Comment (vojtechtrefny) [inline configure.ac:243]: Debian has libnvme 1.3 in testing, we have 1.4 in Fedora 38 and CentOS 9 Stream so I think we might just require 1.3 and have one less check here.

Comment (tbzatek) [reply configure.ac:243]: True, we still haven't released anything and it will take some more time :-)

Comment (tbzatek) [reply configure.ac]: Fixed.

**Comment by tbzatek** [comment]
Jenkins, test this please.

**Pull request merged**